core/src/main/java/software/amazon/awssdk/runtime/StandardMemberCopier.java

```diff
@@ -74,6 +74,19 @@ public static ByteBuffer copy(ByteBuffer bb) {
         if (bb == null) {
             return null;
         }
-        return bb.duplicate();
+
+        ByteBuffer copy = bb.isDirect() ? ByteBuffer.allocateDirect(bb.capacity()) : ByteBuffer.allocate(bb.capacity());
```

Comment:
Sure, I don't have a strong opinion on it. We can stick to a heap buffer all of the time.